packages/webdriverio/src/middlewares.ts

```diff
@@ -24,9 +25,17 @@ export const elementErrorHandler = (fn: Function) => (commandName: string, comma
                  * as `stale element reference`
                  */
                 if (result && result.error === 'no such element') {
-                    const err = new Error()
-                    err.name = 'stale element reference'
-                    throw err
+                    let currentElement = element
+                    //Crawl back to the browser object
+                    while (currentElement.elementId && currentElement.parent) {
+                        currentElement = currentElement.parent as WebdriverIO.Element
+                    }
```

Comment:
There is already a utility method for that: https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/utils/index.ts#L94-L100